be/test/vec/columns/common_column_test.h

```diff
@@ -0,0 +1,766 @@
+            for (auto& column : check_columns) {
+                column->deserialize_vec(input_keys, rows);
+            }
+        }
+        // check the deserialized columns
+        {
+            for (size_t i = 0; i < columns.size(); ++i) {
+                EXPECT_EQ(columns[i]->size(), check_columns[i]->size());
+                checkColumn(*columns[i], *check_columns[i], *dataTypes[i], rows);
+            }
+        }
+    }
+
+    PaddedPODArray<UInt8> create_filter(std::vector<uint8_t> data) {
+        PaddedPODArray<UInt8> filter;
+        filter.insert(filter.end(), data.begin(), data.end());
+        return filter;
+    }
+
+    // filter calculation:
+    //  filter (called in Block::filter_block_internal to filter data with filter filled with 0or1 array, like: [0,1,0,1])
+    //   used in join to filter next block by row_ids, and filter column by row_ids in first read which called in SegmentIterator
+    void filterAssert(MutableColumnPtr col, std::vector<uint8_t> filter, size_t expect_size) {
+        EXPECT_EQ(col->size(), filter.size());
+        auto filted_col = col->filter(create_filter(filter), expect_size);
+        EXPECT_EQ(filted_col->size(), expect_size);
+        int iter = 0;
+        for (size_t i = 0; i < filter.size(); i++) {
+            if (filter[i]) {
+                iter++;
+            }
+        }
+    }
+
+    //  filter_by_selector (called SegmentIterator::copy_column_data_by_selector,
+    //  now just used in filter column, according to the selector to
+    //  select size of row_ids for column by given column, which only used for predict_column and column_dictionary, column_nullable sometimes in Schema::get_predicate_column_ptr() also will return)
+    void filterBySelectorAssert(vectorized::IColumn::MutablePtr col, std::vector<uint16_t> selector,
+                                const IDataType& dt, MutableColumnPtr should_sel_col,
+                                size_t expect_size) {
+        // only used in column_nullable and predict_column, column_dictionary
+        EXPECT_TRUE(col->is_nullable() || col->is_column_dictionary() ||
+                    col->is_predicate_column());
+        // for every data type should assert behavior in own UT case
+        col->clear();
+        col->insert_many_defaults(should_sel_col->size());
+        std::cout << "col size:" << col->size() << std::endl;
+        auto sel = should_sel_col->clone_empty();
+        Status st = col->filter_by_selector(selector.data(), expect_size, sel);
+        EXPECT_EQ(st, Status::OK());
+        EXPECT_EQ(sel->size(), expect_size);
+        printColumn(*sel, dt);
+    }
+
+    void assertPermutationsWithLimit(const IColumn::Permutation& lhs,
+                                     const IColumn::Permutation& rhs, size_t limit) {
+        if (limit == 0) {
+            limit = lhs.size();
+        }
+
+        for (size_t i = 0; i < limit; ++i) {
+            ASSERT_EQ(lhs[i], rhs[i]);
```

Comment:
What does this check for?

Reply:
I think you can just move it inside assertColumnPermutation.

Reply:
done!